cpp/src/binary/jit/core/launcher.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Copyright 2018-2019 BlazingDB, Inc.
+ *     Copyright 2018 Christian Noboa Mardini <christian@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "binary/jit/core/launcher.h"
+#include "binary/jit/code/code.h"
+#include "types.h.jit"
+#include <cstdint>
+
+namespace cudf {
+namespace binops {
+namespace jit {
+
+/**---------------------------------------------------------------------------*
+ * @brief Cache to hold previously compiled code. If JITIFY_THREAD_SAFE is
+ *  defined, then the cache is held globally in the process, otherwise it is
+ *  held locally in each thread
+ * 
+ *---------------------------------------------------------------------------**/
+#ifdef JITIFY_THREAD_SAFE
```

Comment:
Isn't it the thread safe thing to cache thread-locally rather than globally?

Reply:
This define is for Jitify wherein specifying this would cause Jitify to use locks for thread safety. Meaning if Jitify is ensuring thread safety then we can go ahead and keep the cache in global space.